packages/flutter/lib/src/material/carousel.dart

```diff
@@ -389,27 +389,55 @@ class _CarouselViewState extends State<CarouselView> {
     }
   }
 
+  Widget _buildCarouselContent(
+      int index, WidgetStateProperty<Color?>? overlayColor) {
+    if (widget.enableSplash) {
+      return Stack(
+        fit: StackFit.expand,
+        children: <Widget>[
+          widget.children[index],
+          Material(
+            color: Colors.transparent,
+            child: InkWell(
+              onTap: () => widget.onTap?.call(index),
+              overlayColor: overlayColor,
+            ),
+          ),
+        ],
+      );
+    }
+    if (widget.onTap != null) {
+      return GestureDetector(
+        onTap: () => widget.onTap!(index),
+        child: widget.children[index],
+      );
+    }
+    return widget.children[index];
+  }
+
   Widget _buildCarouselItem(ThemeData theme, int index) {
-    final EdgeInsets effectivePadding = widget.padding ?? const EdgeInsets.all(4.0);
-    final Color effectiveBackgroundColor = widget.backgroundColor ?? Theme.of(context).colorScheme.surface;
+    final EdgeInsets effectivePadding =
+        widget.padding ?? const EdgeInsets.all(4.0);
+    final Color effectiveBackgroundColor =
+        widget.backgroundColor ?? Theme.of(context).colorScheme.surface;
     final double effectiveElevation = widget.elevation ?? 0.0;
-    final ShapeBorder effectiveShape = widget.shape
-      ?? const RoundedRectangleBorder(
-        borderRadius: BorderRadius.all(Radius.circular(28.0))
-      );
-    final WidgetStateProperty<Color?> effectiveOverlayColor = widget.overlayColor
-      ?? WidgetStateProperty.resolveWith((Set<WidgetState> states) {
-        if (states.contains(WidgetState.pressed)) {
-          return theme.colorScheme.onSurface.withOpacity(0.1);
-        }
-        if (states.contains(WidgetState.hovered)) {
-          return theme.colorScheme.onSurface.withOpacity(0.08);
-        }
-        if (states.contains(WidgetState.focused)) {
-          return theme.colorScheme.onSurface.withOpacity(0.1);
-        }
-        return null;
-      });
+    final ShapeBorder effectiveShape = widget.shape ??
+        const RoundedRectangleBorder(
+            borderRadius: BorderRadius.all(Radius.circular(28.0)));
+    final WidgetStateProperty<Color?> effectiveOverlayColor =
+        widget.overlayColor ??
+            WidgetStateProperty.resolveWith((Set<WidgetState> states) {
+              if (states.contains(WidgetState.pressed)) {
+                return theme.colorScheme.onSurface.withOpacity(0.1);
+              }
+              if (states.contains(WidgetState.hovered)) {
+                return theme.colorScheme.onSurface.withOpacity(0.08);
+              }
+              if (states.contains(WidgetState.focused)) {
+                return theme.colorScheme.onSurface.withOpacity(0.1);
+              }
+              return null;
+            });
```

Comment:
Also, if you're able to revert these whitespace changes, that'd be awesome.

Reply:
Sure, Sorry, it was my mistake